Apps/Common/src/Authorization/UserAuthorizationHandler.cs

```diff
@@ -15,35 +15,58 @@
 //-------------------------------------------------------------------------
 namespace HealthGateway.Common.Authorization
 {
+    using System.Security.Claims;
     using System.Threading.Tasks;
     using Microsoft.AspNetCore.Authorization;
     using Microsoft.Extensions.Logging;
 
+    /// <summary>
+    /// UserAuthorizationHandler is a custom authorization handler that checks whether the authenticated user has an hdid claim that matches the hdid of the resource being accessed.
+    /// </summary>
     public class UserAuthorizationHandler : AuthorizationHandler<UserIsPatientRequirement, string>
     {
         private readonly ILogger<UserAuthorizationHandler> logger;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UserAuthorizationHandler"/> class.
+        /// </summary>
+        /// <param name="logger">the injected logger.</param>
         public UserAuthorizationHandler(ILogger<UserAuthorizationHandler> logger)
         {
             this.logger = logger;
         }
 
-        protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, UserIsPatientRequirement requirement, string resource)
+        /// <summary>
+        /// HandleRequirementAsync method to assert that the hdid in the JWT matches the hdid in the request.
+        /// </summary>
+        /// <param name="context">the AuthorizationHandlerContext context.</param>
+        /// <param name="requirement">the UserIsPatientRequirement requirement.</param>
+        /// <param name="hdid">The patient identifier used as the resource argument.</param>
+        /// <returns>a Task.</returns>
+        protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, UserIsPatientRequirement requirement, string hdid)
         {
-            string hdidClaim = context?.User.FindFirst(c => c.Type == "hdid").Value;
-
-            if (!string.Equals(hdidClaim, resource, System.StringComparison.Ordinal))
+            if (this.IsOwner(context?.User, hdid))
             {
-#pragma warning disable CA1303 // Do not pass literals as localized parameters
-                this.logger?.LogWarning(@"hdid parameter doest not match user's JWT");
-#pragma warning restore CA1303 // Do not pass literals as localized parameters
-                context?.Fail();
-                return Task.CompletedTask;
+                context?.Succeed(requirement);
             }
 
-            context?.Succeed(requirement);
-
             return Task.CompletedTask;
         }
+
+        /// <summary>
+        /// Check if the authenticated user is the same patient as what is being accessed.
+        /// </summary>
+        /// <param name="user">The authenticated user.</param>
+        /// <param name="hdid">The health data resource subject identifier.</param>
+        private bool IsOwner(ClaimsPrincipal user, string hdid)
+        {
+            if (user == null || hdid == null)
```

Comment:
Is this method ever called with either user or hdid being null?  Instead of the if/return false would a contract be better?  Example: Contract.Requires(hdid != null);